docs/pages/api.rst

```diff
@@ -2133,16 +2062,17 @@ Signature: ``Collection::window(int $size);``
 
 .. code-block:: php
 
-     $data = range('a', 'z');
+     $data = range('a', 'd');
 
      Collection::fromIterable($data)
-        ->window(2)
-        ->all(); // [ ['a'], ['a', 'b'], ['b', 'c'], ['c', 'd'], ... ]
+        ->window(2); // [['a'], ['a', 'b'], ['a', 'b', 'c'], ['b', 'c', 'd']]
```

Comment:
This is the actual output of the method, as also exemplified by the [spec](https://github.com/loophp/collection/blob/master/spec/loophp/collection/CollectionSpec.php#L3223-L3239). 

However, this is really surprising to me. I would expect the following, given that the parameter passed is size of the window (2 in this case).

```PHP
Collection::fromIterable(range('a', 'd'))
    ->window(2); // [['a', 'b'], ['b', 'c'], ['c', 'd']]
```

Removing [this](https://github.com/loophp/collection/blob/master/src/Operation/Window.php#L45) gets it closer to that: `[['a'], ['a', 'b'] , ['b', 'c'], ['c', 'd']`. But I'm still not sure about the first element.

@drupol is this a bug or am I missing something?

Reply:
This is not a bug, basically the `Window` operation "adds" an amount of item, on top of the current one.

This is why if you set it 2, you might end up getting windows of 3 items. (_2 + 1 current_).

Reply:
Hmm I see, I will update the explanation to reflect that. Although the behaviour is still a bit unexpected, because a typical sliding window would only result in groupings of the same size. What's the reason for also having the first element separate?

Reply:
Because it would add opinionated behavior and I'm trying to avoid that. Some people would need them, some would not.

However, if someone doesn't want to see the first elements which are not yet of the size of the sliding window, they can drop them.

```php
$input = array_combine(range('a', 'z'), range('a', 'z'));

$collection = Collection::fromIterable($input);

$output = $collection
    ->window(3)
    ->drop(3);

foreach ($output as $key => $value) {
    var_dump($value); 
}
```

Does it help?

Reply:
Ok I think I get what you mean. 

The behaviour is as follows: `window(int $size)` starts building a window of elements until it gets to the size `1 + $size`, and then keeps moving along the elements of the collection keeping that same size. 

The behaviour I was confused about with the smaller size elements at first makes sense in this context because the window needs to first reach that given size.

Reply:
That's precisely that.